lib/rules/selector-max-attribute/index.mjs

```diff
@@ -92,8 +90,21 @@ const rule = (primary, secondaryOptions) => {
 			if (!isStandardSyntaxRule(ruleNode)) return;
 
 			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector, resolvedSelectors }) => {
-				resolvedSelectors.forEach((resolvedSelector) => {
-					checkSelector(resolvedSelector, selector, ruleNode);
+				resolvedSelectors.walk((childSelector) => {
+					if (childSelector.type !== 'selector') return;
```

Comment:
[suggest] Minor. The `isSelector` type-guard might be a bit more consistent and readable (for example, see `isRoot` and `isPseudo`). But this is not required, you can leave it as-is. 👍🏼 

```suggestion
					if (!isSelector(childSelector)) return;
```